### ISSUE TITLE
libsodium error

### ISSUE THREAD

**Issue by tonikelope** [meganz/MEGAcmd#897]
ArchLinux
Megacmd 1.6.3
Libsodium 1.0.19

```
[Initiating MEGAcmd server in background. Log: /root/.megaCmd/megacmdserver.log]
Unable to connect to service: error=2
Please ensure mega-cmd-server is running
Failed to create socket for registering for state changes
```

Dowgrading libsodium to 1.0.18 solves the error.

Regards.

**Comment by mindorgan** [comment]
MEGAcmd server specifically looks for libsodium.so.23, but the recent version is libsodium.so.26.

So an alternative way is to make a link.

```bash
sudo ln -s /usr/lib/libsodium.so /usr/lib/libsodium.so.23
```

 But downgrading libsodium might be a safer way.

**Comment by paulie-g** [comment]
> But downgrading libsodium might be a safer way.

It's unlikely to be 'safer' in any way, except for making `MEGAcmd` work. There is a reason libsodium developers put out new releases. They are likely to contain improvements, including security improvements, ie more safety, not less. If MEGAcmd continues to work with more recent libsodium, it means the ABI has not changed and using the newer version is a significantly better option. Remember, it's a shared library - downgrading it affects every other program that links to it.

**Comment by mindorgan** [comment]
> They are likely to contain improvements, including security improvements, ie more safety, not less.

You're totally right! I usually go for convenience without thinking much about security. Thanks for the heads up that the new releases usually brings better security and safety, not the other way around. I appreciate your comment 🙏🏻
